datatables.go

```diff
@@ -65,6 +66,81 @@ func (p *Parser) bindEntities() {
 	p.bindBombSites()
 	p.bindPlayers()
 	p.bindWeapons()
+	p.bindBomb()
+}
+
+func (p *Parser) bindBomb() {
+	bombState := p.gameState.bombState
+
+	// Track bomb when it's not held by a player
+	p.stParser.ServerClasses().FindByName("CC4").OnEntityCreated(func(bomb *st.Entity) {
+		bombState.Position = bomb.Position()
+
+		bomb.OnPositionUpdate(func(pos r3.Vector) {
+			bombState.Player = nil // Bomb only has a position when not held by a player
+			bombState.Position = bomb.Position()
+		})
+		bomb.OnDestroy(func() {
+			bombState.Position = bomb.Position()
+		})
+	})
+
+	// Track bomb when has been planted
+	scPlantedC4 := p.stParser.ServerClasses().FindByName("CPlantedC4")
+	scPlantedC4.OnEntityCreated(func(bombEntity *st.Entity) {
+		bombState.Player = nil // Player can't hold bomb when it has been planted
+		bombState.Position = bombEntity.Position()
+	})
+
+	// Track bomb when it's being held by a player
+	scPlayerC4 := p.stParser.ServerClasses().FindByName("CC4")
+	scPlayerC4.OnEntityCreated(func(bombEntity *st.Entity) {
+		wep := p.equipmentMapping[scPlayerC4]
+		if wep != common.EqBomb {
+			return
+		}
+
+		// TODO: fix this hack to "unregister" OnUpdate handlers
+		handlerRegistered := make(map[int]bool)
+
+		bombEntity.FindProperty("m_hPrevOwner").OnUpdate(func(val st.PropertyValue) {
+			ownerEntityID := val.IntVal & entityHandleIndexMask
+			handlerRegistered[ownerEntityID] = false
+		})
+
+		bombEntity.FindProperty("m_hOwner").OnUpdate(func(val st.PropertyValue) {
+			// TODO: for performance, it would be nice to be able to unregister the
+			// potential previous OnUpdate registrations from below, here
+
+			ownerEntityID := val.IntVal & entityHandleIndexMask
+			owner := p.gameState.playersByEntityID[ownerEntityID]
+			if owner == nil {
+				return
+			}
+			handlerRegistered[ownerEntityID] = true
+
+			updateBombPosition := func() {
+				if !handlerRegistered[ownerEntityID] {
+					return
+				}
+
+				if _, hasBomb := owner.RawWeapons[bombEntity.ID()]; !hasBomb {
```

Comment:
Maybe comment why this is needed (I guess it can't be assumed to always be true?)